dataframe/dataframe.go

```diff
@@ -656,6 +657,52 @@ func WithTypes(coltypes map[string]series.Type) LoadOption {
 	}
 }
 
+// LoadStructs creates a new DataFrame from arbitrary []structs
+func LoadStructs(i interface{}, options ...LoadOption) DataFrame {
+	if i == nil {
+		return DataFrame{Err: fmt.Errorf("load: can't create DataFrame from <nil> value")}
+	}
+	tpy, val := reflect.TypeOf(i), reflect.ValueOf(i)
+	switch tpy.Kind() {
+	case reflect.Slice:
+		if tpy.Elem().Kind() != reflect.Struct {
+			return DataFrame{Err: fmt.Errorf("load: type %s (%s %s) is not supported, must be []struct", tpy.Name(), tpy.Elem().Kind(), tpy.Kind())}
+		}
+		if val.Len() == 0 {
+			return DataFrame{Err: fmt.Errorf("load: can't create DataFrame from empty slice")}
+		}
+		structs := []interface{}{}
+		for i := 0; i < val.Len(); i++ {
+			structs = append(structs, val.Index(i).Interface())
+		}
+		maps := []map[string]interface{}{}
+		for i := range structs {
+			strcTpy, strcVal := reflect.TypeOf(structs[i]), reflect.ValueOf(structs[i])
+			temp := map[string]interface{}{}
+		NextField:
+			for i := 0; i < strcTpy.NumField(); i++ {
+				if strcTpy.Field(i).Anonymous {
+					continue NextField
+				}
+				switch strcVal.Field(i).Kind() {
```

Comment:
In order for this to not panic you have to check if it is possible or not to get the interface:

```go
if strcTpy.Field(i).Anonymous || !strcVal.Field(i).CanInterface() {
	continue
}
temp[strcTpy.Field(i).Name] = strcVal.Field(i).Interface()
```

This make it so that unexported struct fields are ignored, which is also the default behaviour of `encoding/json` making it more predictable.